shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java

```diff
@@ -142,4 +168,27 @@ private void processingFailed(Class<?> entity, Throwable t) {
     private void processingComplete(Class<?> entity) {
         LOGGER.debug("[shard] Successfully processed all entities '{}' to deploy or delete", entity.getSimpleName());
     }
+
+    private Uni<HttpResponse<Buffer>> getAuthenticatedRequest(HttpRequest<Buffer> request, Function<HttpRequest<Buffer>, Uni<HttpResponse<Buffer>>> executor) {
+        if (client.isPresent()) {
+            return Uni.createFrom().deferred(() -> {
+                tokenLock.lock();
```

Comment:
I have to admit this section was challenging to an old school guy. 
I assume we need to syncrhonize on tokens to avoid multiple token creation (to keep the token unique for all request from the shard to the manager)  but at the same time I do not want the reactive thing to get block. 
This is what it came out, deferring the creation of the Uni till the suscribe and block if the access token has expired till is renewed. 
Please note that isAccessTokenExpired is pretty fast call so normally there will not blocking. 
Also I take the decission to intialize the token for the first time the first time is used and not at initialization time.

Reply:
@fjtirado I wonder if the `Tokens` producer takes care of this for you already. It would seem to suggest that is the case from the Quarkus documentation.

Reply:
what I understand from documentation is that it does not.